kore/src/Kore/Parser/Lexeme.hs

```diff
@@ -217,36 +229,44 @@ idCharSet :: CharSet
 idCharSet =
     CharSet.join idFirstCharSet (CharSet.makeCharSet idOtherChars)
 
-{-|'objectIdRawParser' extracts the string representing an @object-identifier@.
-Does not consume whitespace.
--}
-objectIdRawParser :: IdKeywordParsing -> Parser String
-objectIdRawParser keywordParsing = do
-    backslash <- Parser.string "\\" <|> pure ""
-    name <- objectNonslashIdRawParser keywordParsing
-    pure (backslash ++ name)
-
-objectNonslashIdRawParser :: IdKeywordParsing -> Parser String
-objectNonslashIdRawParser =
+symbolIdRawParser :: Parser String
+symbolIdRawParser = do
```

Comment:
Why not,

```.hs
symbolIdRawParser =
    ((:) <$> char '\\' <*> genericKoreIdRawParser KeywordsPermitted)
    <|> genericKoreIdRawParser KeywordsForbidden
```

(Forgiving my formatting.)

Reply:
Maybe we could rewrite more of our parsing infrastructure this way;  This was a way to clearly show that you only have 1 char lookahead. In your definition, this is harder to grasp.

Reply:
That's true. It's a trade-off: the current implementation shows look-ahead very clearly versus the `Alternative` definition can be made to look (almost exactly) like the syntax document. We should keep it as-is to be consistent.